src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/ProjectSystem/WorkspaceProjectStateChangeDetectorTest.cs

```diff
@@ -107,14 +110,17 @@ public async Task WorkspaceChanged_SolutionEvents_EnqueuesUpdatesForProjectsInSo
             var workspaceStateGenerator = new TestProjectWorkspaceStateGenerator();
             var detector = new WorkspaceProjectStateChangeDetector(workspaceStateGenerator, Dispatcher);
             var projectManager = new TestProjectSnapshotManager(Dispatcher, new[] { detector }, Workspace);
-            projectManager.ProjectAdded(HostProjectOne);
-            projectManager.ProjectAdded(HostProjectTwo);
+            await Dispatcher.RunOnForegroundAsync(() =>
+            {
+                projectManager.ProjectAdded(HostProjectOne);
+                projectManager.ProjectAdded(HostProjectTwo);
+            }, CancellationToken.None);
 
             var e = new WorkspaceChangeEventArgs(kind, oldSolution: EmptySolution, newSolution: SolutionWithTwoProjects);
 
             // Act
-            detector.Workspace_WorkspaceChanged(Workspace, e);
-            await Dispatcher.ForegroundScheduler; // Wait for event to finish processing on thread
+            await Dispatcher.RunOnForegroundAsync(() => detector.Workspace_WorkspaceChanged(Workspace, e), CancellationToken.None);
+            await Dispatcher.ForegroundScheduler;
```

Comment:
This is super weird and I'm not sure if it's the best approach here, but it works (at least from the couple of times I ran it) - otherwise the assert below will execute before the Workspace_WorkspaceChanged has finished.

Reply:
Edit: jk, it doesn't work - exploring alternatives now